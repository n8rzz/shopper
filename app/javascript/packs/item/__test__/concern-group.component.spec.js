import React from 'react';
import {shallow} from 'enzyme';

import {ConcernGroup} from '../concern-group.component';
import {
    departmentMapMock,
    itemListMock
} from './__mock__/item.mocks';

describe('ConcernGroup', () => {
    test('renders correctly with valid props', () => {
        const component = shallow(<ConcernGroup
            csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
            departmentMap={departmentMapMock}
            groupName={"Bakery"}
            itemList={itemListMock}
        />);

        expect(component).toMatchSnapshot();
    });
});
