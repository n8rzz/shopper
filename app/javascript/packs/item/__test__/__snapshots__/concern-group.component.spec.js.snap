// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConcernGroup renders correctly with valid props 1`] = `
<div
  className="concernGroup"
  key="concernGroup-Bakery"
>
  <div
    className="concernGroup-hd"
  >
    <h2>
      Bakery
    </h2>
  </div>
  <ItemList
    csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
    departmentMap={
      Object {
        "3": Object {
          "created_at": "2019-07-21T16:03:32.883Z",
          "id": 3,
          "name": "Bakery",
          "updated_at": "2019-07-21T16:03:32.883Z",
        },
        "5": Object {
          "created_at": "2019-07-21T16:03:32.883Z",
          "id": 5,
          "name": "Meat",
          "updated_at": "2019-07-21T16:03:32.883Z",
        },
      }
    }
    itemList={
      Array [
        Object {
          "created_at": "2019-07-21T16:03:33.162Z",
          "department_id": 5,
          "id": 15,
          "name": "Agave Syrup",
          "updated_at": "2019-07-21T16:03:33.162Z",
        },
        Object {
          "created_at": "2019-07-21T16:03:33.162Z",
          "department_id": 3,
          "id": 13,
          "name": "Agave Syrup",
          "updated_at": "2019-07-21T16:03:33.162Z",
        },
      ]
    }
  />
</div>
`;
